sdk/core/azure-core/src/main/java/com/azure/core/util/polling/PollerFlux.java

```diff
@@ -217,8 +217,12 @@ public void subscribe(CoreSubscriber<? super AsyncPollResponse<T, U>> actual) {
             () -> this.rootContext.copy(),
             // Do polling
             // set|read to|from context as needed, reactor guarantee thread-safety of cxt object.
-            cxt -> Mono.defer(() -> this.pollOperation.apply(cxt))
-                .delaySubscription(getDelay(cxt.getLatestResponse()))
+            cxt -> Mono.defer(() -> {
+                final Mono<PollResponse<T>> pollOnceMono = this.pollOperation.apply(cxt);
+                // Execute (subscribe to) the pollOnceMono after the default poll-interval
+                // or duration specified in the last retry-after response header elapses.
+                return pollOnceMono.delaySubscription(getDelay(cxt.getLatestResponse()));
+            })
```

Comment:
The issue was incorrect scoping of `getDelay` method.

The `getDelay()` method called once subscription on outer `Flux` is applied. This result in whatever value that call to `getDelay` returns (say 3 sec)  gets hardcoded in `delaySubscription`, it looks like:
 
```java
Mono.defer(() -> this.pollOperation.apply(cxt))
    .delaySubscription(3)
    .switchIfEmpty(Mono.error(new IllegalStateException("PollOperation returned Mono.empty().")))
    .repeat()
```
 
then when we `repeat`, this fixed value applied instead of a new call to `getDelay` to retrieve the new value.